web_src/js/features/repo-graph.js

```diff
@@ -18,27 +18,27 @@ export function initRepoGraphGit() {
       window.history.replaceState({}, '', window.location.pathname);
     }
     $('.pagination a').each((_, that) => {
-      const href = $(that).attr('href');
+      const href = that.href;
       if (!href) return;
       const url = new URL(href, window.location);
       const params = url.searchParams;
       params.set('mode', 'monochrome');
       url.search = `?${params.toString()}`;
-      $(that).attr('href', url.href);
+      that.href = url.href;
```

Comment:
Use `setAttribute` instead of `href`. There are slight differences in those two methods in how the URL encodes as I recently noticed.

Reply:
Ah I see @wxiaoguang suggests the opposite.

Reply:
Could you elaborate, what's the difference? https://developer.mozilla.org/en-US/docs/Web/API/HTMLAnchorElement/href

Reply:
@wxiaoguang see https://github.com/go-gitea/gitea/pull/29931#issuecomment-2010363267. There I came to the conclusion it's better to use `setAttribute` and `getAttribute` when working with `href`.

Reply:
I do not think so. `el.href` is the normalized and correctly escaped full URL. It could just be used as-is.

But `getAttribute("..")` just returns the raw string value, it could be incorrectly escaped value.

Reply:
In your example, `a.setAttribute("href", "#气候宜人")` you are setting a invalid URL to `href`. So when you read the `el.href` again, it gets correctly escaped.

Reply:
You could try this:

```
const a = document.createElement("a");
a.setAttribute("href", "/what ever bad url");
a.textContent = "link";
document.body.append(a);
console.log(a.href);  --> https://fiddle.jshell.net/what%20ever%20bad%20url , correctly escaped
console.log(a.getAttribute("href"));  --> /what ever bad url  ,  still the bad URL
```


ps: I am just talking about the details. For this PR, any approach could work, I am neutral.

Reply:
> In your example, `a.setAttribute("href", "#气候宜人")` you are setting a invalid URL to `href`. So when you read the `el.href` again, it gets correctly escaped.

It's not a invalid URL as `URL` constructor accepts it:

```
new URL("#气候宜人", "https://example.com");
{
  href: 'https://example.com/#%E6%B0%94%E5%80%99%E5%AE%9C%E4%BA%BA',
  origin: 'https://example.com',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'example.com',
  hostname: 'example.com',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: '#%E6%B0%94%E5%80%99%E5%AE%9C%E4%BA%BA'
}
```

Reply:
We should just do what jQuery did. @yardenshoham could you debug it out please?

Reply:
Yes, it essentially calls `getAttribute`.
https://github.com/jquery/jquery/blob/f79d5f1a337528940ab7029d4f8bbba72326f269/src/attributes/attr.js#L68
https://github.com/jquery/jquery/blob/f79d5f1a337528940ab7029d4f8bbba72326f269/src/selector.js#L875

Reply:
Yeah so let's change it back so it's a more accurate refactor?

Reply:
> > In your example, `a.setAttribute("href", "#气候宜人")` you are setting a invalid URL to `href`. So when you read the `el.href` again, it gets correctly escaped.
> 
> It's not a invalid URL as `URL` constructor accepts it:

URL also generates the normalized and fully escaped URL: `href: 'https://example.com/#%E6%B0%94%E5%80%99%E5%AE%9C%E4%BA%BA'`. So I insist `el.href` is the right thing to be directly used.

Reply:
Both will work I assume. That said, this code looks ripe for refactor, these two functions are almost completely the same.